drivers/instr-amplif/ada4250/ada4250.c

```diff
@@ -0,0 +1,304 @@
+{
+	int32_t ret;
+	uint8_t buff[ADA4250_BUFF_SIZE_BYTES];
+
+	buff[0] = ADA4250_SPI_READ_CMD | reg_addr;
+	buff[1] = ADA4250_SPI_DUMMY_DATA;
+
+	ret = spi_write_and_read(dev->spi_desc, buff, ADA4250_BUFF_SIZE_BYTES);
+	if(ret != SUCCESS)
+		return ret;
+
+	*data = buff[1];
+
+	return ret;
+}
+
+/**
+ * @brief Update ADA4250 register.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - Mask for specific register bits to be updated.
+ * @param data - Data read from the device.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+ */
+int32_t ada4250_update(struct ada4250_dev *dev, uint8_t reg_addr,
+		       uint8_t mask, uint8_t data)
+{
+	uint8_t read_val;
+	int32_t ret;
+
+	ret = ada4250_read(dev, reg_addr, &read_val);
+	if (ret != SUCCESS)
+		return ret;
+
+	read_val &= ~mask;
+	read_val |= data;
+
+	ret = ada4250_write(dev, reg_addr, read_val);
+
+	return ret;
+}
+
+/**
+ * @brief Software reset.
+ * @param dev - The device structure.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t ada4250_soft_reset(struct ada4250_dev *dev)
+{
+	int32_t ret;
+	uint16_t timeout = 0xFFFF;
+	uint8_t data;
+
+	ret = ada4250_update(dev, ADA4250_REG_RESET, ADA4250_RESET_MSK,
+			     ADA4250_RESET(ADA4250_RESET_ENABLE));
+
+	while(timeout--) {
+		ret = ada4250_read(dev, ADA4250_REG_RESET, &data);
+		if (ret != SUCCESS)
+			return ret;
+
+		if(!(data & ADA4250_RESET(ADA4250_RESET_ENABLE)))
+			return SUCCESS;
+	}
+
+	return FAILURE;
+}
+
+/**
+ * @brief Enable/Disable Reference Buffer.
+ * @param dev - The device structure.
+ * @param refbuf - REFBUF enable/disable.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t ada4250_set_refbuf(struct ada4250_dev *dev, uint8_t refbuf)
+{
+	return ada4250_update(dev, ADA4250_REG_REFBUF_EN, ADA4250_REFBUF_MSK,
+			      ADA4250_REFBUF(refbuf));
+}
+
+/**
+ * @brief Set current bias for ADA4250.
+ * @param dev - The device structure.
+ * @param bias - Current bias option.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t ada4250_set_bias(struct ada4250_dev *dev, uint8_t bias)
+{
+	return ada4250_update(dev, ADA4250_REG_SNSR_CAL_CNFG, ADA4250_BIAS_SET_MSK,
+			      ADA4250_BIAS_SET(bias));
+}
+
+/**
+ * @brief Set offset trim range for ADA4250.
+ * @param dev - The device structure.
+ * @param range - Offset range option.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t ada4250_set_range(struct ada4250_dev *dev, uint8_t range)
+{
+	return ada4250_update(dev, ADA4250_REG_SNSR_CAL_CNFG, ADA4250_RANGE_SET_MSK,
+			      ADA4250_RANGE_SET(ADA4250_RANGE(range)));
+}
+
+/**
+ * @brief Set gain for ADA4250.
+ * @param dev - The device structure.
+ * @param gain - Gain Value.
+ * @return Returns SUCCESS in case of success or negative error code.
+ */
+int32_t ada4250_set_gain(struct ada4250_dev *dev, uint8_t gain)
```

Comment:
This is not very intuitive either: the possible actual gain values are 1, 2, 4, 8, 16, 32, 64, 128.

Reply:
Addressed with: https://github.com/analogdevicesinc/no-OS/pull/984